src/cgame/cg_main.cpp

```diff
@@ -989,7 +989,7 @@ void CG_RegisterGrading( int slot, const char *str )
 		return;
 	}
 
-	sscanf(str, "%d %f %s", &model, &dist, texture);
+	sscanf(str, "%d %f %63s", &model, &dist, texture);
```

Comment:
Better not to hard-code the size. You can use `%*s` and pass the number of characters size in an argument.

Reply:
Mmm, not sure about this.

From http://www.cplusplus.com/reference/cstdio/scanf/:
> * starting asterisk indicates that the data is to be read from the stream  but ignored (i.e. it is not stored in the location pointed by an argument).

Originally i was thinking about this:

```
#define STR2(x) #x
#define STR(X) STR2(X)

#define MAXVAL 5

char string[MAXVAL + 1];
scanf("%" STR(MAXVAL) "s", string);
```

but it looks like such macroses should be stored in daemon, since they may be useful in general. And i don't know either we will update daemon to the latest version or not.

Reply:
https://stackoverflow.com/questions/25410690/scanf-variable-length-specifier

Reply:
My mistake, apparently it does not work the same way as printf.

The macro way seems not so nice either due to the need to add/subtract 1. However if you wish to pursue it, the stringification macros are [here](https://github.com/DaemonEngine/Daemon/blob/master/src/engine/qcommon/q_shared.h#L173).

I'd say it's OK to merge in the current form with hard-coded sizes. All the code in the Unvanquished repo normally runs in a secure sandbox, so we can have lower standards for secure coding than the code in Daemon.